src/main/kotlin/com/sourcegraph/cody/listeners/CodySelectionInlayManager.kt

```diff
@@ -0,0 +1,162 @@
+
+                // Right edge
+                path.lineTo(
+                    (targetRegion.x + targetRegion.width).toDouble(),
+                    (targetRegion.y + targetRegion.height - arcSize).toDouble())
+                path.quadTo(
+                    (targetRegion.x + targetRegion.width).toDouble(),
+                    (targetRegion.y + targetRegion.height).toDouble(),
+                    (targetRegion.x + targetRegion.width - arcSize).toDouble(),
+                    (targetRegion.y + targetRegion.height).toDouble())
+
+                // Bottom edge
+                path.lineTo(
+                    (targetRegion.x + arcSize).toDouble(),
+                    (targetRegion.y + targetRegion.height).toDouble())
+                path.lineTo(
+                    targetRegion.x.toDouble(), (targetRegion.y + targetRegion.height).toDouble())
+
+                // Left edge
+                path.lineTo(targetRegion.x.toDouble(), targetRegion.y.toDouble())
+
+                path.closePath()
+                (g as Graphics2D).fill(path)
+
+                val descent = g.fontMetrics.descent
+                val textColor = editor.colorsScheme.getColor(EditorColors.CARET_COLOR)
+                g.color = textColor
+
+                val baseline = targetRegion.y + targetRegion.height - descent - 4
+                g.drawString(content, targetRegion.x, baseline)
+                inlayBounds = targetRegion
+              }
+            })
+
+    inlay?.let {
+      Disposer.register(disposable, it)
+      currentInlay = it
+    }
+  }
+
+  private fun clearInlay() {
+    currentInlay?.let { Disposer.dispose(it) }
+    currentInlay = null
+    inlayBounds = null
+  }
+
+  @Suppress("SameParameterValue")
+  private fun getKeyStrokeText(actionId: String): String {
+    val shortcuts = KeymapManager.getInstance().activeKeymap.getShortcuts(actionId)
+    if (shortcuts.isNotEmpty()) {
+      val firstShortcut = shortcuts[0]
+      if (firstShortcut is KeyboardShortcut) {
+        val keyStroke = firstShortcut.firstKeyStroke
+        val modifiers = keyStroke.modifiers
+        val key = KeyEvent.getKeyText(keyStroke.keyCode)
+        val isMac = System.getProperty("os.name").lowercase(Locale.getDefault()).contains("mac")
+        val separator = " + "
+
+        val modText = buildString {
+          append(" ") // Add some separation from the end of the source code.
+          if (modifiers and KeyEvent.CTRL_DOWN_MASK != 0) append("Ctrl$separator")
+          if (modifiers and KeyEvent.SHIFT_DOWN_MASK != 0) append("Shift$separator")
+          if (modifiers and KeyEvent.ALT_DOWN_MASK != 0) append("Alt$separator")
+          if (isMac && modifiers and KeyEvent.META_DOWN_MASK != 0) append("Cmd$separator")
+        }
+        return (modText + key).removeSuffix(separator)
+      }
+    }
+    return ""
+  }
+
+  fun handleSelectionChanged(editor: Editor, event: SelectionEvent) {
+    val startOffset = event.newRange.startOffset
```

Comment:
nit: I see there are two execution branches here
1. startOffset == endOffset -> clearInlay -> return
2. startOffset != endOffset -> updateInlay -> clearInlay -> set new inline

IIUC you can extract `clearInlay()` and call it at the beginning of this method.

Reply:
Fixed!